packages/drivers/local-driver/src/localDocumentDeltaConnection.ts

```diff
@@ -42,130 +38,23 @@ export class LocalDocumentDeltaConnection
-                    // add them to the list of initialMessages to be processed
-                    response.initialMessages.push(...queuedMessages);
-                    response.initialMessages.sort((a, b) => a.sequenceNumber - b.sequenceNumber);
-                }
-
-                if (queuedSignals.length > 0) {
-                    // Some signals were queued.
-                    // add them to the list of initialSignals to be processed
-                    response.initialSignals.push(...queuedSignals);
-                }
-
-                resolve(response);
-            });
-
-            socket.on("connect_document_error", reject);
-
-            socket.emit("connect_document", connectMessage);
-        });
-
-        const deltaConnection = new LocalDocumentDeltaConnection(socket, id, connection);
-
-        return Promise.resolve(deltaConnection);
-    }
-
-    private readonly submitManager: BatchManager<IDocumentMessage[]>;
-    private readonly subscribedEvents = new Set<string>();
-
-    public get clientId(): string {
-        return this.details.clientId;
-    }
-
-    public get mode(): ConnectionMode {
-        return this.details.mode;
-    }
-
-    public get claims(): ITokenClaims {
-        return this.details.claims;
+        await deltaConnection.initialize(connectMessage, timeoutMs);
+        return deltaConnection;
     }
 
-    public get existing(): boolean {
-        return this.details.existing;
-    }
-
-    public get maxMessageSize(): number {
-        return this.details.maxMessageSize;
-    }
-
-    public get initialMessages(): ISequencedDocumentMessage[] {
-        return this.details.initialMessages;
-    }
-
-    public get initialSignals(): ISignalMessage[] {
-        return this.details.initialSignals;
-    }
-
-    public get initialClients(): ISignalClient[] {
-        return this.details.initialClients;
-    }
-
-    public get version(): string {
-        return testProtocolVersions[0];
-    }
-
-    public get serviceConfiguration(): IClientConfiguration {
-        return this.details.serviceConfiguration;
-    }
-
-    constructor(
-        private readonly socket: core.IWebSocket,
-        public documentId: string,
-        public details: IConnected) {
-        super();
-
-        this.submitManager = new BatchManager<IDocumentMessage[]>((submitType, work) => {
-            this.socket.emit(
-                submitType,
-                this.details.clientId,
-                work,
-                (error) => {
-                    if (error) {
-                        debug("Emit error", error);
-                    }
-                });
-        });
+    constructor(socket: LocalWebSocket, documentId: string) {
+        // Cast LocalWebSocket to SocketIOClient.Socket which is the socket that the base class needs. Doing this
+        // should be fine because this delta connection is used for local use only.
+        const socketWithListener = socket as unknown as SocketIOClient.Socket;
```

Comment:
Any chance we push these requirements / assumptions up the stack? I see caller does another cast, it would be great to push these casts out to consumers

Reply:
I can move it to the create function. But I don't think it can be moved up any further.